src/Components/Recommendations/BookRec.js

```diff
@@ -0,0 +1,244 @@
+/*
+    ============ TODO =============
+    - Get current book's tags
+    - query airtable for a list of similar books
+    - store the list somewhere
+
+    Prioritization:
+      primary -> age-range, grade-range, race/ethnicity, and identity
+      secondary -> author,  genre, theme/lessons
+
+    Filter:
+      must make sure that the same book is not shown
+
+    How to make sure prioritization is in the correct range:
+      - priorisation: books that have the most correct matches to the top tier.
+      - top tier: age_range, grade_range, race/ethnicity
+      - if books have same number of correct matches,
+        prioritize age_range > grade_range > race_ethnicity
+
+    Challenges:
+      - Tweak filter that can work with array of data
+      - Tweak filter to work with range of values
+
+    If airtable filter doesn't work:
+      fetch all books from airtable
+      apply filters and increment the priority counter for each book accordingly
+      sort based on priority and display the top n books
+
+*/
+
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import BookCards from './BookRecCards';
+
+// airtable configuration
+const Airtable = require('airtable');
+
+const airtableConfig = {
+  apiKey: process.env.REACT_APP_AIRTABLE_USER_KEY,
+  baseKey: process.env.REACT_APP_AIRTABLE_BASE_KEY,
+};
+
+const base = new Airtable({ apiKey: airtableConfig.apiKey })
+  .base(airtableConfig.baseKey);
+
+export default function RecCardsDisplay({
+  minAge, maxAge, minGrade, maxGrade, raceEthnicity,
+}) {
+  const [book, setBook] = useState([]);
+  const gradeList = ['0 to Pre-K', 'Kindergarten', '1st', '2nd', '3rd', '4th', '5th', '6th', '7th', '8th', '9th', '10th', '11th', '12th'];
+  const prioMap = new Map();
+  const [recList, setRecList] = useState([]);
+  // const [ageRange, setAgeRange] = useState([]);
+  // const [gradeRange, setGradeRange] = useState([]);
+  /*
+  const [ethnicity, setEthnicity] = useState([]); */
+  // const [identity, setIdentity] = useState();
+  // const [filteredRec, setFilteredRec] = useState([]);
+
+  // const getCards = () => {
+  //   base('Book').select({ view: 'Grid view' }).all()
+  //     .then((records) => {
+  //       setBook(records);
+  //     });
+  // };
+
+  // Instead of using props, we pull bookId from URL
+  const { bookId } = useParams();
+
+  // Airtable Filter implementation
+  // filterByFormula: `IF(${field} = {id},
+  // SEARCH("search for age_range", "id"), IF("", "", SEARCH("${fieldKeyword}", {${field}})))`
+  /*
+IF(
+        ${field} = {id},
+        SEARCH("search for age_range", "id"),
+        IF("", "", SEARCH("${fieldKeyword}", {${field}}))
+      )
+  */
+  // console.log({ minAge });
+  // console.log({ maxAge });
+  // console.log({ minGrade });
+  // console.log({ maxGrade });
+  // console.log({ raceEthnicity });
+
+  function RecFilter(field, fieldKeyword) {
```

Comment:
it doesn't seem like this function does anything with these two parameters! You can probs just get rid of them